src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Scanner;
+
+public class Storage {
+    private Path filePath = null;
+    private Path directoryPath = null;
+
+    /**
+     * Constructor for a new Storage object.
+     *
+     * @param path Path for the new storage object.
+     */
+    public Storage(String path) {
+        String home = System.getProperty("user.home");
+        this.filePath = java.nio.file.Paths.get(home, "iP", path.split("/")[0], path.split("/")[1]);
+        this.directoryPath = java.nio.file.Paths.get(home, "iP", path.split("/")[0]);
+    }
+
+    /**
+     * Converts Task object to a text format suitable for saving in a storage.
+     *
+     * @param task Task object to be converted to text.
+     * @return String representation of the given task.
+     */
+    public String convertTaskToText(Task task) {
+        String result = task.getType() + "|";
+        if (task.isCompleted()) {
+            result += "1|";
+        } else {
+            result += "0|";
+        }
+        result += task.getTaskContent();
+        if (task.getType().equals("D") || task.getType().equals("E")) {
+            result += "|" + task.getTiming();
+        }
+        return result;
+    }
```

Comment:
Consider using String.format() here, I think it enhances readability and saves the concatenation.